acceptancetests/deploy_stack.py

```diff
@@ -137,24 +147,75 @@ def deploy_dummy_stack(client, charm_series, use_charmstore=False):
         client.wait_for_started(3600)
 
 
-def deploy_caas_stack(bundle_path, client, timeout=3600):
+def _deploy_stack(path, client, timeout=3600, charm=False, lxd_profile=None):
     # workaround to ensure lxd profile
-    model_name = client.model_name
-    profile = LXD_PROFILE.format(model_name=model_name)
-    with subprocess.Popen(('echo', profile), stdout=subprocess.PIPE) as echo:
-        o = subprocess.check_output(
-            ('lxc', 'profile', 'edit', 'juju-%s' % model_name),
-            stdin=echo.stdout
-        ).decode('UTF-8').strip()
-        log.debug(o)
-
-    client.deploy_bundle(bundle_path, static_bundle=True)
+    if lxd_profile is not None:
+        model_name = client.model_name
+        profile = lxd_profile.format(model_name=model_name)
+        with subprocess.Popen(('echo', profile), stdout=subprocess.PIPE) as echo:
+            o = subprocess.check_output(
+                ('lxc', 'profile', 'edit', 'juju-%s' % model_name),
+                stdin=echo.stdout
+            ).decode('UTF-8').strip()
+            log.debug(o)
+
+    # Deploy a charm or a bundle depending on the flag
+    if not charm:
+        client.deploy_bundle(path, static_bundle=True)
+    else:
+        client.deploy(charm=path)
     # Wait for the deployment to finish.
     client.wait_for_started(timeout=timeout)
     # wait for cluster to stabilize
-    client.wait_for_workloads()
+    client.wait_for_workloads(timeout=timeout)
     # get current status with tabular format for better debugging
     client.juju(client._show_status, ('--format', 'tabular'))
+    return client
+
+
+def deploy_iaas_stack(path, client, timeout=3600, charm=False):
+    """deploy a IAAS stack, it assumes that the path to a bundle/charm can be
+       reached.
+
+       When deploying the stack, it will use a LXD profile to inject into the
+       lxc client, so that we can take advantage of nested containers, for
+       example.
+
+    :param path: location/path to the bundle/charm
+    :param client: client used to deploy and watch bundle/charm workloads
+    :param timeout: timeout of when the workload should fail deployment
+    :param charm: deploy a charm or bundle
+    """
+
+    client = _deploy_stack(
+        path, client,
+        timeout=timeout,
+        charm=charm,
+        lxd_profile=IAAS_LXD_PROFILE,
+    )
+    return IaasClient(client)
+
+
+def deploy_caas_stack(path, client, timeout=3600, charm=False):
```

Comment:
I see, the CAAS name I think is really off here. Deploying a k8s cluster isn't really CAAS as that's related to the k8s charms work that's gone on. If it's what's there though this is an ok backport.

Reply:
I just copied the files over from develop to make sure there is no conflict when we merge 2.5 -> develop.